src/IssueViz.Security.UnitTests/HotspotsList/HotspotsTableDataSourceTests.cs

```diff
@@ -164,6 +164,111 @@ public void Add_HasSubscribersToIssuesChangedEvent_SubscribersNotified()
             suppliedArgs.AnalyzedFiles.Should().BeEquivalentTo("a.cpp", "b.cpp");
         }
 
+        [TestMethod]
+        public void Remove_NoMatchingEntries_IssuesChangedEventNotRaised()
+        {
+            var issueViz = CreateIssueViz("a.cpp");
+
+            var testSubject = CreateTestSubject();
+            testSubject.Add(issueViz);
+
+            var eventCount = 0;
+            testSubject.IssuesChanged += (sender, args) => { eventCount++; };
+
+            testSubject.Remove(CreateIssueViz("b.cpp"));
+
+            eventCount.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void Remove_NoMatchingEntries_SinksNotNotified()
+        {
+            var issueViz = CreateIssueViz("a.cpp");
+
+            var testSubject = CreateTestSubject();
+            testSubject.Add(issueViz);
+
+            var sink = new Mock<ITableDataSink>();
+            testSubject.Subscribe(sink.Object);
+
+            sink.Reset();
+
+            testSubject.Remove(CreateIssueViz("b.cpp"));
+
+            sink.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Remove_HasMatchingEntries_IssuesChangedEventRaised()
+        {
+            var issueViz1 = CreateIssueViz("a.cpp");
+            var issueViz2 = CreateIssueViz("b.cpp");
+
+            var testSubject = CreateTestSubject();
+            testSubject.Add(issueViz1);
+            testSubject.Add(issueViz2);
+
+            IssuesChangedEventArgs suppliedArgs = null;
+            var eventCount = 0;
+            testSubject.IssuesChanged += (sender, args) => { suppliedArgs = args; eventCount++; };
+
+            testSubject.Remove(issueViz1);
+
+            eventCount.Should().Be(1);
+            suppliedArgs.Should().NotBeNull();
+            suppliedArgs.AnalyzedFiles.Should().BeEquivalentTo("a.cpp");
+        }
+
+        [TestMethod]
+        public void Remove_HasMatchingEntries_SinksNotified()
+        {
+            var issueViz1 = CreateIssueViz("a.cpp");
+            var issueViz2 = CreateIssueViz("b.cpp");
+
+            var testSubject = CreateTestSubject();
+            testSubject.Add(issueViz1);
+            testSubject.Add(issueViz2);
+
+            var sink = new Mock<ITableDataSink>();
+            testSubject.Subscribe(sink.Object);
+
+            sink.Reset();
+
+            testSubject.Remove(issueViz1);
```

Comment:
It doesn't hurt to vary the test inputs between otherwise identical tests.
In all three of the "remove with matches" tests you're removing the first item in the list each time. Logically, these tests would still all pass if the product code only ever checked the first item in the list i.e. the tests don't demonstrate that the product code iterates through the list of entries.

You could add a separate test for that case, but that can get pretty tedious. In any event, the general point stands that by varying the data you might catch something you weren't expecting.